src/goto-instrument/contracts/instrument_spec_assigns.cpp

```diff
@@ -519,6 +519,44 @@ car_exprt instrument_spec_assignst::create_car_expr(
           upper_bound_var,
           car_havoc_methodt::HAVOC_SLICE};
       }
+      else if(ident == CPROVER_PREFIX "object_whole")
+      {
+        const auto &ptr = funcall.arguments().at(0);
+        return {
+          condition,
+          target,
+          minus_exprt(
+            typecast_exprt::conditional_cast(ptr, pointer_type(char_type())),
+            pointer_offset(ptr)),
+          typecast_exprt::conditional_cast(object_size(ptr), size_type()),
```

Comment:
Nitpick: a comment above this expression would be helpful. Same below.

Reply:
+1